### DISCUSSION TITLE
bug: [Process exited 0] when using gitblame

### DISCUSSION THREAD

**Discussion by dmmulroy** [folke/snacks.nvim#323] [Q&A]
### Did you check docs and existing issues?

- [X] I have read all the snacks.nvim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of snacks.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

v0.10.2

### Operating system/version

MacOS 14.7.1 (23H222)

### Describe the bug

When loading the git blame window, it fails to be able to scroll or show a full diff and instead loads/shows part of the diff followed by `[Process exited 0]`. Not quite sure how to debug the failure -  i suspect this is a me issue and not a `snack.nvim` issue.

![image](https://github.com/user-attachments/assets/15c5e33f-3d4a-4303-9d92-e7d54c4f5fc6)

```sh
commit 81eaf439f64d6903f215af18e674293e9cfb975d
Author: Dillon Mulroy <dillon.mulroy@gmail.com>
Date:   Thu Mar 14 10:11:52 2024 -0400

    merge nvim

diff --git a/config/nvim/lua/plugins/lsp.lua b/config/nvim/lua/plugins/lsp.lua
--- a/config/nvim/lua/plugins/lsp.lua
+++ b/config/nvim/lua/plugins/lsp.lua
@@ -62,0 +98,1 @@
+                               html = {},

[Process exited 0]
```

snacks config:

```lua
local macchiato = require("catppuccin.palettes").get_palette("macchiato")
vim.api.nvim_set_hl(0, "SnacksIndentScope", { fg = macchiato.mauve })

return {
	{
		"folke/snacks.nvim",
		priority = 1000,
		lazy = false,
		---@type snacks.Config
		opts = {
			bigfile = { enabled = true },
			git = { enabled = true },
			gitbrowse = { enabled = true },
			indent = { enabled = true },
			notifier = { enabled = true },
			quickfile = { enabled = true },
			statuscolumn = { enabled = true },
			terminal = { enabled = true },
			words = { enabled = true },
			zen = { enabled = true },
		},
	},
}
```

### Steps To Reproduce

1. Open the gitblame utility anywhere 

### Expected Behavior

it allows me to page up/down without exiting

### Repro

```lua
Will try to create a minimal repro this evening
```

**Comment by lisiur** [comment]
if you are using `delta` as pager, add this to your `~/.gitconfig`

```
[delta]
    paging = always
```

> The reason is that if there is only one page, `git log` will simply print the content and exit. However, when `pager = always` is set, `git log` will always use a pager, meaning it will not exit immediately.

FYI this is my keymap config:

```lua
vim.keymap.set({'n'}, '<leader>B', function()
  Snacks.git.blame_line({
    start_insert = true,
    auto_close = true,
  })
end, {desc = "Blame line"})
```

**Comment by 0111b** [comment]
Same issue with dashboard terminal section. After launch it shows "[Process exited 0]" with any cmd.

MacOS 26.4.1  NVIM v0.12.2 snacks.nvim ad9ede6

<img width="603" height="107" alt="Screenshot 2026-04-27 at 16 32 07" src="https://github.com/user-attachments/assets/1c3ba4f0-c177-4762-89bc-0cbdb4a8d136" />
